core/store/src/trie/trie_storage.rs

```diff
@@ -119,7 +119,7 @@ impl TrieStorage for TrieMemoryPartialStorage {
 
 /// Maximum number of cache entries.
 #[cfg(not(feature = "no_cache"))]
-const TRIE_MAX_CACHE_SIZE: usize = 10000;
```

Comment:
Yes, I think ```    Self(Arc::new(Mutex::new(SizedCache::with_size(TRIE_MAX_CACHE_SIZE)))) ```, could be the problem. 

Replacing it with `SyncLruCache` could help a little bit https://github.com/near/nearcore/pull/5632
Once, this gets approved, I'll work on even better version, as there are a few other issues with it, that could cause big issues.

It is theoretically, possible, that the current implementation of `TrieCache` causes the issues.
I have a plan to fix it, but I need to wait for https://github.com/near/nearcore/pull/5632 to be approved first.

To improve:
- use SyncLruCache, that uses a better implemented `LruCache` instead of `cached`.
- Instead of using single `Mutex`, use a shared mutex, so we have for example, 16 smaller caches, and the chances of hitting a mutex that's busy, are much lower. Latency, spikes can happen, if potentially, a thread is swapped out, while hitting a mutex, etc.
- There, are a few other issues with the cache, I can fix them, the limitation is lack of reviews ...

Reply:
Anyway, that's a bit out of topic, not blocking this PR. I think increase cache size is an excellent idea!